flow/project.py

```diff
@@ -3199,6 +3199,12 @@ def run(
             )
         if names is None:
             names = list(self.operations)
+        else:
+            absent_ops = (set(self._groups.keys()) ^ set(names)) & set(names)
+            if absent_ops:
+                print(
+                    f"The requested flow operation(s) does not exist: {', '.join(absent_ops)}"
+                )
```

Comment:
We can't perform this check yet. At this point, the `names` are regular expressions that can match groups. There's a line below here (3287) where these regular expressions are converted to a list of matching operations.

```python
run_groups = set(self._gather_flow_groups(names))
```

What we probably need to do is check below and show a warning if the provided expressions match zero group names. Try this with operations named `op1`, `op2` and an expression like `project.py run -o op`.

Reply:
Oh wait, sorry. I see that `_gather_flow_groups` was also edited. If that's correct, then why do we need the check here?

Reply:
Could we just edit `_gather_flow_groups` instead?

Reply:
@bdice We do a `re.fullmatch` for the names. This means performing either the way I did or moving the code to `_gather_flow_groups` would do the same thing. The reason I was not willing to move it to `_gather_flow_groups` was that since we iterate over line [`3287`](https://github.com/glotzerlab/signac-flow/blob/feature/warnings-for-non-existent-op/flow/project.py#L3287) multiple times in `run`, we'd then end up printing the message multiple times.